configs/typescript.js

```diff
@@ -28,6 +28,9 @@ module.exports = {
     // checking such as indexed array iteration
     "@typescript-eslint/no-non-null-assertion": "off",
 
+    // Printing Error variables into template strings is a common pattern
```

Comment:
I don't think this justification alone is very good...you can always call `toString()` on the Error. Is that the only use case you ran into? It would be nice if the lint rule could determine whether the object has a toString method, but that doesn't seem to be a feature.

Reply:
You can only catch variables of type `any`, so there is no `toString()` method that TypeScript is aware of. You can wrap everything in `String()` but I don't think there's a good justification for doing that.

Reply:
What if we were to enable the rule with all the [options](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-template-expressions.md) (allowNumber, allowBoolean, allowAny, allowNullish) set to true? Would that still correctly prevent objects like arrays / object literals from being used?

Reply:
Why do we want to ban this?

```
const arr = [1, 2, 3];
console.log(`arr is ${arr}`);
```
> arr is 1,2,3

Reply:
I guess that's useful for console logs but I don't think we would ever want this to show up in the UI anywhere (with no space after the comma)...